storage/encrypting_repository.go

```diff
@@ -0,0 +1,172 @@
+package storage
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"time"
+
+	"github.com/Peripli/service-manager/pkg/log"
+
+	"github.com/Peripli/service-manager/pkg/security"
+
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+)
+
+func NewEncryptingRepository(ctx context.Context, repository securedTransactionalRepository, encrypter security.Encrypter) (*EncryptingRepository, error) {
```

Comment:
This interface is needed so that we can pass a repository that is secured and to not cast to Secured. We can export, or we can remove this interface and pass two arguments (Secured and TransactionalRepository) but we will pass the same struct smStroage for both). Or we can pass just TransactionaRepository and cast.